### PR TITLE
fix(anim): Fix the problem that the animation time is shortened  #3572

### PR DESCRIPTION
**Pull request by qinshijing**
Signed-off-by: qinshijing <qinshijing@xiaomi.com>

### Description of the feature or fix

A clear and concise description of what the bug or new feature is.

### Checkpoints
- [x] Follow the [styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [x] Run `code-format.py` from the `scripts` folder. [astyle](http://astyle.sourceforge.net/install.html) needs to be installed.
- [ ] Update the documentation

### CODE REVIEW

**Comment by qinshijing** [comment]
Hello, I pushed a fix, please review it, thank you.

File: src/misc/lv_anim.c [hunk 1]
```diff
@@ -402,13 +397,16 @@ static void anim_timer(lv_timer_t * param)
 
         /*If the linked list changed due to anim. delete then it's not safe to continue
          *the reading of the list from here -> start from the head*/
-        if(anim_list_changed)
+        if(anim_list_changed) {
             a = _lv_ll_get_head(&LV_GC_ROOT(_lv_anim_ll));
-        else
+        }
+        else {
+            a->last_timer_run = lv_tick_get();
```
Comment (kisvegabor) [inline src/misc/lv_anim.c]: I'd move it right below ` uint32_t elaps = lv_tick_elaps(a->last_timer_run);`
Else when the list changes for any reason `last_timer_run` won't be updated.

**Review by kisvegabor — COMMENTED** [review]
Looks good, thank you. 
I have only one comment.

**Comment by kisvegabor** [comment]
Great, thank you!

**Pull request merged**